services/secondary-storage/server/src/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java

```diff
@@ -525,6 +525,8 @@ protected Answer copySnapshotToTemplateFromNfsToNfs(CopyCommand cmd, SnapshotObj
                     bufferWriter.write("\n");
                     long size = _storage.getSize(destFileFullPath);
                     bufferWriter.write("size=" + size);
+                    bufferWriter.close();
```

Comment:
@rhtyd : We already have try-with-resource declaration for the bufferWriter. But here we need to explicitly close this bufferWriter because in below code within try block , call is being made to save() method of TemplateLocation class where one more bufferWriter comes into picture for writing few other properties in Template.properties file.